### PR TITLE
go: Apply the options and correct the bitmasks

### PR DESCRIPTION
**Pull request by qinqon**
The golang binding was not applying the option that were passed and also the bitmask was not correct since it has to start from "2". It also create a pair of apply test for the golang binding and activate the logs at the clib.

Depends on:
- https://github.com/nmstate/nmstate/pull/1811

### CODE REVIEW

File: rust/src/clib/lib.rs [hunk 1]
```diff
@@ -167,7 +170,7 @@ pub extern "C" fn nmstate_net_state_apply(
 
     net_state.set_timeout(rollback_timeout);
 
-    // TODO: save log to the output pointer
+    let _ = env_logger::try_init();
```
Comment (cathay4t) [inline rust/src/clib/lib.rs]: This just pring logs to stderr which should not be done by a C library.
As the API designed, the log should saved to `char *`.

Initial plan is mimic what was done by https://github.com/gahag/memory_logger/

File: rust/src/clib/nmstate.h.in [hunk 2]
```diff
@@ -186,92 +186,20 @@ int nmstate_checkpoint_rollback(const char *checkpoint, char **log, char **err_k
...
- *
- * Return:
- *      void
- */
-void nmstate_checkpoint_free(char *checkpoint);
-
-#ifdef __cplusplus
-} /* extern "C" */
-#endif
+void nmstate_cstring_free(char *cstring);
```
Comment (cathay4t) [inline rust/src/clib/nmstate.h.in]: This is unrelated to this PR.
Unless good reasoning, I don't like to the C API/ABI.
Current functions sames good to me.

File: rust/src/go/nmstate/Makefile [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
 
 build:
 	go build;
 
+check: TMPDIR:=$(shell mktemp -d)
 check:
-	go test;
-
+	DESTDIR=$(TMPDIR) make -C $(TOP_SRC_DIR) install
+	sudo RUST_LOG=info LD_LIBRARY_PATH=$(TMPDIR)/usr/local/lib64 CGO_CFLAGS="-I$(TMPDIR)/usr/local/include" CGO_LDFLAGS="-L$(TMPDIR)/usr/local/lib64" go test $(WHAT);
```
Comment (cathay4t) [inline rust/src/go/nmstate/Makefile]: Try to make text width 78.

**Review by cathay4t — CHANGES_REQUESTED** [review]
The go code is reviewed as I am not familiar it yet.

File: rust/src/clib/lib.rs (hunk 1 above)
Comment (qinqon) [reply rust/src/clib/lib.rs]: Ahh that's the logger I was looking for, thanks! I will try to integrate it.

File: rust/src/clib/nmstate.h.in (hunk 2 above)
Comment (qinqon) [reply rust/src/clib/nmstate.h.in]: This PR depends on https://github.com/nmstate/nmstate/pull/1795, I can put common code under a new functions and convert the current functions as wrappers that call it.

File: rust/src/clib/lib.rs (hunk 1 above)
Comment (qinqon) [reply rust/src/clib/lib.rs]: @cathay4t Looks like the tokio-rs projects has a tracing library with non-global tracer https://github.com/tokio-rs/tracing, you can define the scope of a trace and run code there.

Maybe  nmstate rust can use that library don't know if we can make it use memory or retrieve what has being logged.

Comment (cathay4t) [reply rust/src/clib/lib.rs]: The tokio tracing seems overkill for this project. Please try memory logger. But this can be done in another PR. For this PR, just remove the print in stdout.

**Comment by cathay4t** [comment]
Please enable CI test for go binding since you included the test case.

**Comment by qinqon** [comment]
> Please enable CI test for go binding since you included the test case.

Calling `make -C rust check` at last commit.

File: rust/src/clib/test/Makefile [hunk 4]
```diff
@@ -1,36 +1,42 @@
...
-libnmstate.so: $(DEBUG_NMSTATE_SO_FILE)
-	install -p -v -D -m755 $(DEBUG_NMSTATE_SO_FILE) \
+install: $(CLIB_SO_DEV_DEBUG)
+	install -p -v -D -m755 $(CLIB_SO_DEV_DEBUG) \
 		$(TMPDIR)/$(CLIB_SO_FULL)
 	ln -sfv $(CLIB_SO_FULL) $(TMPDIR)/$(CLIB_SO_MAN)
 	ln -sfv $(CLIB_SO_FULL) $(TMPDIR)/$(CLIB_SO_DEV)
 
-nmstate_test: nmstate_test.c nmstate.h libnmstate.so
+nmstate_test: nmstate_test.c nmstate.h install
```
Comment (cathay4t) [inline rust/src/clib/test/Makefile]: Please use `CGO_LDFLAGS` and `CGO_CFLAGS` instead of install it. I ran `make check` in my OS which I don't want it been installed in this case

**Review by cathay4t — CHANGES_REQUESTED** [review]

File: rust/src/clib/test/Makefile (hunk 4 above)
Comment (qinqon) [reply rust/src/clib/test/Makefile]: I will copy (although it introduces some redundancy) the build/install code the the src/go/nmstate/Makefile

**Review by cathay4t — APPROVED** [review]
